monai/networks/nets/torchvision_fc.py

```diff
@@ -0,0 +1,43 @@
+import torch
+
+from monai.utils import optional_import
+
+models, _ = optional_import("torchvision.models")
+
+
+class TorchVisionFullyConvModel(torch.nn.Module):
+    """
+    Customize TorchVision models to replace fully connected layer by convolutional layer.
+
+    Args:
+        model_name: name of any torchvision with adaptive avg pooling and fully connected layer at the end.
+            Default to "resnet18".
+        n_classes: number of classes for the last classification layer. Default to 1.
+        pretrained: whether to use the imagenet pretrained weights. Default to False.
+    """
+
+    def __init__(self, model_name: str = "resnet18", n_classes: int = 1, pretrained: bool = False):
+        super().__init__()
+        model = getattr(models, model_name)(pretrained=pretrained)
+
+        # remove last fully connected (FC) layer and adaptive avg pooling
+        self.features = torch.nn.Sequential(*list(model.children())[:-2])
+
+        # add 7x7 avg pooling (in place of adaptive avg pooling)
+        self.pool = torch.nn.AvgPool2d(7, stride=1)
+
+        # add 1x1 conv (it behaves like a FC layer)
+        self.fc = torch.nn.Conv2d(model.fc.in_features, n_classes, kernel_size=(1, 1))
```

Comment:
perhaps if the n_class is the same as the pretrained, we should also initialise the conv kernels using the pretrained model fc layers?

Reply:
this could be left for future work

Reply:
It's an easy fix but I thought to treat different class numbers equally. Also in MONAI people would use 1000 classes rarely if at all.

Reply:
sure, agreed, I didn't know that all of models are of 1000 classes